### DISCUSSION TITLE
Allow to send a list of solvers

### DISCUSSION THREAD

**Discussion by erelsgl** [cvxpy/cvxpy#1522] [Ideas]
In some problems, some solvers may be better than others, but they might also fail with a `SolverError`.  I suggest allow, as the "solver" parameter, to send a list of solvers rather than just one. The function will try them in the order they are given, and return the first solution for which no `SolverError` is raised.

**Comment by SteveDiamond** [comment]
Agreed, we should do this.

**Comment by erelsgl** [comment]
Here is an example code. The code gets as inputs the problem to solve and a list of tuples. Each tuple contains the name of the solver, and the keyword arguments to send to that solver.

```
def solve(problem:cvxpy.Problem, solvers:List[Tuple[str, Dict]]):
	is_solved=False
	for (solver,kwargs) in solvers:  # Try the first n-1 solvers.
		try:
			problem.solve(solver=solver, **kwargs)
			logger.info("Solver %s succeeds",solver)
			is_solved = True
			break
		except cvxpy.SolverError as err:
			logger.info("Solver %s fails: %s", solver, err)
	if not is_solved:
		raise cvxpy.SolverError(f"All solvers failed: {solvers}")

```
